drivers/i2c/CMakeLists.txt

```diff
@@ -16,15 +16,14 @@ zephyr_library_sources_ifdef(CONFIG_I2C_SAM_TWI		i2c_sam_twi.c)
 zephyr_library_sources_ifdef(CONFIG_I2C_SAM_TWIHS	i2c_sam_twihs.c)
 zephyr_library_sources_ifdef(CONFIG_I2C_SBCON		i2c_sbcon.c)
 zephyr_library_sources_ifdef(CONFIG_I2C_NIOS2		i2c_nios2.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V1        i2c_ll_stm32_v1.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V2        i2c_ll_stm32_v2.c)
 
-zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V1
-	i2c_ll_stm32_v1.c
-	i2c_ll_stm32.c
-	)
-zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V2
-	i2c_ll_stm32_v2.c
-	i2c_ll_stm32.c
-	)
+zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V1        i2c_ll_stm32.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V2        i2c_ll_stm32.c)
+
+zephyr_library_sources_jinjagen_ifdef(CONFIG_I2C_STM32_V1 i2c_ll_stm32_device.h.jinja2)
+zephyr_library_sources_jinjagen_ifdef(CONFIG_I2C_STM32_V2 i2c_ll_stm32_device.h.jinja2)
```

Comment:
Are these last 2 lines really mandatory?

Can't we just call zephyr_library_sources_jinjagen_ifdef with the .c file of the driver we want to apply generation on?
So here would be:
```
zephyr_library_sources_jinjagen_ifdef(CONFIG_I2C_STM32_V1        i2c_ll_stm32.c)
zephyr_library_sources_jinjagen_ifdef(CONFIG_I2C_STM32_V2        i2c_ll_stm32.c)
```
So that developer only needs to care about its driver filename and that's it. Once we know the name of the file on which we want to apply device generation, things are pretty mechanical. We can build up the device filename template 'i2c_ll_stm32_device.h.jinja2'.
Makes things a bit clearer and minimize potential errors, you can't mix up between
- Use zephyr_library_sources_jinjagen_ifdef for devices generation, or
- Use zephyr_library_sources_ifdef

Besides, it is also a way to impose the format and limit the possible names variations that will pop up if we let freedom to the users. Also, having uniform names should also be helpful for various tools that we might want to use on them (I have no idea what tools yet, but we always end up using tools).

Reply:
Yes. Your proposal can be implemented.

If the RFC is accepted we'll get this done.